### DISCUSSION TITLE
Multiple edge handles per node

### DISCUSSION THREAD

**Discussion by bogdanbiv** [cytoscape/cytoscape.js#2834] [Q&A]
Part of the functionality I'd want would be similar to https://github.com/cytoscape/cytoscape.js-edgehandles.
However, I did not understand from the source code:
 * Could a node have multiple edge-handles, for different types of edges?
 * Could the edges originate && terminate on the (center of the) edge handles?
 
Would this be possible at all /with with the current EH extension/ or with a new extension ?
UPDATE, for some more context:
I would need this for setting a graph describing a Visual Flow programming where a graph node may have multiple input and outputs. Input and output could be set to be of a specific type and some sort of validation would be performed for compatibility between output and input.

Any two cents/ comments/ ideas?

**Comment by Bokomoko** [comment]
Answering the first question : Yes! A node can have multiple edges "arriving" and/or "departing" from it. The limitation is just visual. Too many edges coming from/going to a node will appear very heavy.

Edges always start and end at a node. If you want anything connected to the middle of an edge it must be a node. You can add different types of nodes to your graph. 

Suppose that you have and edge C connecting nodes A to B.

You want a new connection to node D from the middle of C . YOu have to add a node midway A/B, connect this node with 3 edges, one from A, another for B and a third for D. This node can be styled in a different shape to differentiate it from A and B
